### PR TITLE
Adding Kibana

### PR DESCRIPTION
**Pull request by Elkozel**
## Description
This provides two new scripts for Kibana, widely used for system and security monitoring. The added scripts are:

- ct/kibana.sh
- install/kibana-install.sh

Description about the files are below.

### ct/kibana.sh
The file provides a typical installation, based on the `adguard.sh` installation script provided almost without any changes.

### install/kibana-install.sh
This script follows the `n8n-install.sh` script, completing almost the same steps for installation using `apt-get`. 

## Fixes
No issues were created for this improvement.

## Type of change

Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Documentation update required (this change requires an update to the documentation)

### CODE REVIEW

File: ct/kibana.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/ELKozel/Proxmox/main/misc/build.func)
```
Comment (tteck) [inline ct/kibana.sh:2]: Wrong

File: ct/kibana.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# Copyright (c) 2024 ELKozel
```
Comment (tteck) [inline ct/kibana.sh:3]: Wrong

File: ct/kibana.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+# Author: T.H. (ELKozel)
```
Comment (tteck) [inline ct/kibana.sh:4]: Wrong

File: ct/kibana.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+function update_script() {
+header_info
+if [[ ! -f /etc/systemd/system/Kibana.service ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+if (( $(df /boot | awk 'NR==2{gsub("%","",$5); print $5}') > 80 )); then
+  read -r -p "Warning: Storage is dangerously low, continue anyway? <y/N> " prompt
+  [[ ${prompt,,} =~ ^(y|yes)$ ]] || exit
+fi
+
+msg_info "Stopping ${APP}"
```
Comment (tteck) [inline ct/kibana.sh:63]: Why stop and start being it's an `APT` upgrade?

File: install/kibana-install.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y wget
```
Comment (tteck) [inline install/kibana-install.sh:17]: `wget`is preinstalled in Debian 12
`curl` is missing

File: install/kibana-install.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+msg_info "Setting up Elastic Repository"
+mkdir -p /etc/apt/keyrings
+wget -qO - https://artifacts.elastic.co/GPG-KEY-elasticsearch | gpg --dearmor -o /usr/share/keyrings/elasticsearch-keyring.gpg
+echo "deb [signed-by=/usr/share/keyrings/elasticsearch-keyring.gpg] https://artifacts.elastic.co/packages/8.x/apt stable main" >/etc/apt/sources.list.d/elastic-8.x.list
+msg_ok "Set up Elastic Repository"
+
+msg_info "Installing Kibana"
+$STD apt-get update
+$STD apt-get install kibana
```
Comment (tteck) [inline install/kibana-install.sh:33]: fully functional without the yes flag?

File: install/kibana-install.sh [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+msg_info "Creating Service"
+cat <<EOF >/etc/systemd/system/Kibana.service
+[Unit]
+Description=Kibana
+Documentation=https://www.elastic.co/guide/en/kibana/8.15/index.html
+Wants=network-online.target
+After=network-online.target
+[Service]
+Type=simple
+User=kibana
```
Comment (tteck) [inline install/kibana-install.sh:45]: Is the User & Group setup during the install?

File: install/kibana-install.sh [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y wget
+$STD apt-get install -y sudo
+$STD apt-get install -y mc
+$STD apt-get install -y ca-certificates
```
Comment (tteck) [inline install/kibana-install.sh:20]: Is this needed?

**Comment by tteck** [comment]
Close, but not ready yet. Sorry

File: install/kibana-install.sh (hunk 6 above)
Comment (Elkozel) [reply install/kibana-install.sh:33]: somehow, yes

File: install/kibana-install.sh (hunk 7 above)
Comment (Elkozel) [reply install/kibana-install.sh:45]: Yes, just checked it, this is the `/etc/passwd` after a clean install:
```
root@kibana:~# cat /etc/passwd
root:x:0:0:root:/root:/bin/bash
daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
bin:x:2:2:bin:/bin:/usr/sbin/nologin
sys:x:3:3:sys:/dev:/usr/sbin/nologin
sync:x:4:65534:sync:/bin:/bin/sync
games:x:5:60:games:/usr/games:/usr/sbin/nologin
man:x:6:12:man:/var/cache/man:/usr/sbin/nologin
lp:x:7:7:lp:/var/spool/lpd:/usr/sbin/nologin
mail:x:8:8:mail:/var/mail:/usr/sbin/nologin
news:x:9:9:news:/var/spool/news:/usr/sbin/nologin
uucp:x:10:10:uucp:/var/spool/uucp:/usr/sbin/nologin
proxy:x:13:13:proxy:/bin:/usr/sbin/nologin
www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
backup:x:34:34:backup:/var/backups:/usr/sbin/nologin
list:x:38:38:Mailing List Manager:/var/list:/usr/sbin/nologin
irc:x:39:39:ircd:/run/ircd:/usr/sbin/nologin
_apt:x:42:65534::/nonexistent:/usr/sbin/nologin
nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
systemd-network:x:998:998:systemd Network Management:/:/usr/sbin/nologin
postfix:x:100:109::/var/spool/postfix:/usr/sbin/nologin
sshd:x:101:65534::/run/sshd:/usr/sbin/nologin
systemd-timesync:x:997:997:systemd Time Synchronization:/:/usr/sbin/nologin
messagebus:x:102:111::/nonexistent:/usr/sbin/nologin
kibana:x:103:112::/nonexistent:/bin/false
```

**Pull request closed without merging**